### ISSUE TITLE
pandas gbq

### ISSUE THREAD

**Issue by skrawcz** [DAGWorks-Inc/hamilton#375]
Implement https://pandas.pydata.org/docs/reference/io.html#google-bigquery as a data loader & saver  - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_gbq.html#pandas.DataFrame.to_gbq.

1. It should go into pandas_extensions.py, one class for the writer, on class for the reader, e.g. much like how the [pandas pickle reader and writer](https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/plugins/pandas_extensions.py) are structured.
2. There should be requisite tests to exercise the functionality. Though this will be need to be mocked, since we don't want to connect to GCP to run unit tests. 
3. We will want a separate example for this one. Since it requires having a GCP account.
4. If there's an issue with type hints, let me know and we can chat through it.

**Comment by bryangalindo** [comment]
working on this

**Comment by bryangalindo** [comment]
@skrawcz for the third requirement, assuming we're talking about the materialization example, where do you want to store the example? (e.g., `examples/pandas/materialization/gbq`, `examples/pandas/gbq/materialization`)

**Comment by skrawcz** [comment]
Yeah let's just put it under `examples/pandas/materialization/gbq` and have similar script & notebook ? That way it can have its own README too.

**Comment by bryangalindo** [comment]
I am working on this 👍 

> Yeah let's just put it under `examples/pandas/materialization/gbq` and have similar script & notebook ? That way it can have its own README too.

I'm working on this 👍
